azure-pipelines.yml

```diff
@@ -0,0 +1,68 @@
+trigger:
+  batch: true
+  branches:
+    include:
+      - main
+
+pr:
+  branches:
+    include:
+      - main
+
+schedules:
+  - cron: '0 0 * * 0'
+    displayName: Weekly build
+    branches:
+      include:
+        - main
+    always: true
+
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: osisoft/OSI-Samples
+      endpoint: osisoft/OSI-Samples
+
+variables:
+  - template: '/miscellaneous/build_templates/variables.yml@templates'
+  - name: CredFile
+    value: config.yml
+  - name: CredFileLibrary
+    value: product-readiness.data_views_r.r
+  - name: ComparisonFile
+    value: sampleoutput.csv
+  - name: ComparisonFileLibrary
+    value: product-readiness.ocs_data_views_r.output_comparison.csv
+  - name: analysisProject
+    value: Data_Views_R
+
+jobs:
+  - job: Tests
+    pool:
+      name: 00-OSIManaged-Test
+      demands: COMPUTERNAME -equals $(buildAgent)
+    steps:
+      - template: '/miscellaneous/build_templates/config.yml@templates'
+        parameters:
+          fileOriginal: $(CredFile)
+          fileLibrary: $(CredFileLibrary)
+
+      - template: '/miscellaneous/build_templates/config.yml@templates'
+        parameters:
+          fileOriginal: $(ComparisonFile)
+          fileLibrary: $(ComparisonFileLibrary)
+
+      - script: Rscript.exe run_tests.R
+        displayName: 'Run tests'
+
+      - task: PublishTestResults@2
+        condition: succeededOrFailed()
+        inputs:
+          testResultsFormat: 'JUnit'
+          testResultsFiles: '**\output.xml'
+        displayName: 'Publish test results'
+
+      - template: '/miscellaneous/build_templates/analysis.yml@templates'
```

Comment:
Does BlackDuck actually detect anything for R? I think it doesn't, in which case we should consider just removing this from the pipeline.

Reply:
I don't think it's scanning anything either. The project was created by dev ops, though, and the R Web API sample has the Black Duck scan and project (also seems empty), so I left it in to match it (plus the test technically passes, so it wasn't anything to do, per se). Would you recommend taking it out? It feels safer, from a procedural level, to leave it in and empty since it doesn't fail anything.